query.py

```diff
@@ -156,24 +221,24 @@ def __str__(self):
 
 
 class QueryNot(Query):
-    def __init__(self, op1:Query):
+    def __init__(self, op: Query):
         super().__init__()
-        self.operand1 = op1
-        self.is_flipped = not op1.is_flipped
-        # new_op = op1.copy()
-        # new_op.is_flipped = not new_op.is_flipped
-        # return new_op
-
-    def evaluate(self, inverted_index:InvertedIndex):
-        # TODO remove temporary naive implementation
-        matches = self.operand1.evaluate(inverted_index)
-        return inverted_index.all_files.difference(matches)
+        self.is_primitive = op.is_primitive
+        self.op = op
+        self.is_flipped = not op.is_flipped
+
+    def evaluate(self, inverted_index: InvertedIndex, forced=False):
+        matches = self.op.evaluate(inverted_index)
+        if forced:
+            return list(inverted_index.all_files.difference(matches))
+        else:
```

Comment:
Can add a field self.size.

Similar to QueryOr and QueryNot, before we return the value, set the value for self.size.

def get_size() will then return self.size.